File: E-Shop-Final/src/main/java/ajc/sopra/eshop/model/Category.java
package ajc.sopra.eshop.model;

import java.util.ArrayList;
import java.util.List;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;

import com.fasterxml.jackson.annotation.JsonView;

@Entity
public class Category {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@JsonView(JsonViews.Common.class)
	private Integer id;
	@Column(/*columnDefinition = "VARCHAR(35) default 'pas de category'"*/length=35,nullable = false)
	@JsonView(JsonViews.Common.class)
	private  String labelCat;
	

	@JsonView(JsonViews.CategoryWithProduct.class)
	@OneToMany(mappedBy="category")
	private List<Product> products=new ArrayList(); 
	
	//probleme return null
	public List<Product> getProducts() {
		return products;
	}

	public void setProducts(List<Product> products) {
		this.products = products;
	}

	public Category( String labelCat) {
		this.labelCat = labelCat;
	}
	
	public void setId(Integer id) {
		this.id = id;
	}
	public Category() {
		
	}
	
	public String getLabelCat() {
		return labelCat;
	}
	public void setLabelCat(String labelCat) {
		this.labelCat = labelCat;
	}
	@Override
	public String toString() {
		return "Category [ labelCat=" + labelCat + "]";
	}
	public Integer getId() {
		
		return id;
	}

}

File: E-Shop-Angular/src/app/eshop/model/category.ts
import { Product } from './product';

export class Category {
  public get products(): Product[] | undefined {
    return this._products;
  }
  public set products(value: Product[] | undefined) {
    this._products = value;
  }
  public get labelcat(): string | undefined {
    return this._labelcat;
  }
  public set labelcat(value: string | undefined) {
    this._labelcat = value;
  }
  public get id(): number | undefined {
    return this._id;
  }
  public set id(value: number | undefined) {
    this._id = value;
  }
  constructor(
    private _id?: number,
    private _labelcat?: string,
    private _products?: Product[]
  ) {}
}
